packages/asset-did/src/AssetDid.ts

```diff
@@ -0,0 +1,81 @@
+/**
+ * Copyright (c) 2018-2022, BOTLabs GmbH.
+ *
+ * This source code is licensed under the BSD 4-Clause "Original" license
+ * found in the LICENSE file in the root directory of this source tree.
+ */
+
+import type {
+  AssetDidUri,
+  Caip19AssetId,
+  Caip19AssetInstance,
+  Caip19AssetNamespace,
+  Caip19AssetReference,
+  Caip2ChainId,
+  Caip2ChainNamespace,
+  Caip2ChainReference,
+} from '@kiltprotocol/types'
+import { SDKErrors } from '@kiltprotocol/utils'
+
+// Matches AssetDIDs as per the [AssetDID specification](https://github.com/KILTprotocol/spec-asset-did).
+const ASSET_DID_REGEX =
+  /^did:asset:(?<chainId>(?<chainNamespace>[-a-z0-9]{3,8}):(?<chainReference>[-a-zA-Z0-9]{1,32}))\.(?<assetId>(?<assetNamespace>[-a-z0-9]{3,8}):(?<assetReference>[-a-zA-Z0-9]{1,64})(:(?<assetInstance>[-a-zA-Z0-9]{1,78}))?)$/
+
+type IAssetDidParsingResult = {
+  uri: AssetDidUri
+  chainId: Caip2ChainId
+  chainNamespace: Caip2ChainNamespace
+  chainReference: Caip2ChainReference
+  assetId: Caip19AssetId
+  assetNamespace: Caip19AssetNamespace
+  assetReference: Caip19AssetReference
+  assetInstance?: Caip19AssetInstance
+}
+
+/**
+ * Parses an AssetDID uri and returns the information contained within in a structured form.
+
+ * @param assetDidUri An AssetDID uri as a string.
+* @returns Object containing information extracted from the AssetDID uri.
+ */
+export function parse(assetDidUri: AssetDidUri): IAssetDidParsingResult {
+  const matches = ASSET_DID_REGEX.exec(assetDidUri)?.groups
+  if (!matches) {
+    throw new SDKErrors.InvalidDidFormatError(assetDidUri)
+  }
+
+  const {
+    chainId,
+    chainNamespace,
+    chainReference,
+    assetId,
+    assetNamespace,
+    assetReference,
+    assetInstance,
+  } = matches as Omit<IAssetDidParsingResult, 'did'>
+
+  return {
+    chainId,
+    chainNamespace,
+    chainReference,
+    assetId,
+    assetNamespace,
+    assetReference,
+    assetInstance,
+    uri: `did:asset:${chainId}.${assetId}`,
+  }
```

Comment:
Now that I see it, I think this can be even simpler
```suggestion
  return {
    ...(matches as Omit<IAssetDidParsingResult, 'uri'>),
    uri: `did:asset:${chainId}.${assetId}`,
  }
```

Reply:
We can't use `chainId` and `assetId` in this way 😁

Reply:
I only captured `chanId` and `assetId` and then used your suggestion here. Will push soon.

Reply:
Something like this: https://github.com/KILTprotocol/sdk-js/pull/677/commits/589c8bc7d8f0fbdf240c2824a70adb34102dbfb0#diff-78f6deb4434716917618adc7d5660fc049921f43f0f6e539ea4702316353c594R47